File: 2022/01/1.py
def sum_elf(e):
  return sum(map(int, e.split("\n")))

with open("input.txt") as f:
  elves = sorted(map(sum_elf, f.read().split("\n\n")))

print("Part 1:", elves[-1])
print("Part 2:", sum(elves[-3:]))

File: 2022/01/1.clj
(ns aoc.2022.01.1 
  (:require [aoc.common :refer [read-input sum]]
            [clojure.string :as str]))

(let [elves (->> (read-input :split-with #"\n\n")
                 (map #(str/split % #"\n"))
                 (map #(map parse-long %))
                 (map sum)
                 (sort >))]

  (println "Part 1:" (first elves))
  (println "Part 2:" (sum (take 3 elves))))
